go/app/main.go

```diff
@@ -25,18 +40,74 @@ func root(c echo.Context) error {
 	return c.JSON(http.StatusOK, res)
 }
 
+func dataJson() Items{
+	//Open our jsonFile
+	jsonFile, err := os.Open("items.json")
+	if err != nil {
+		fmt.Println(err)
+	}
+	//Defer the closing of our jsonFile so that we can parse it later on
+	defer jsonFile.Close()
+
+	//Read our opened jsonFile as a byte array.
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+
+	//Inicialize our array of items
+	var beforeItems Items
+
+	//Save data into the array
+	json.Unmarshal(byteValue, &beforeItems)
+
+	return beforeItems
+}
+
 func addItem(c echo.Context) error {
-	// Get form data
+	//Get form data
 	name := c.FormValue("name")
-	c.Logger().Infof("Receive item: %s", name)
+	category := c.FormValue("category")
+	imagePath := c.FormValue("image")
+	//Read the data of the image
+	imageData, _ := ioutil.ReadFile(imagePath)
+
+	//Create new image name with sha256
+    newImageName := fmt.Sprintf("%x%s", sha256.Sum256(imageData), ".jpg")
```

Comment:
Go has an automatic formatting command that we should always use. It is very convenient because many rules about code style have already been decided by it. You can read more about it by searching about `go fmt`.

Reply:
Hi, I am not really sure how should I improve this part. I read about go fmt in the go.dev website but I am not sure how to apply it here. Could you guide me a little bit more? Thank you

Reply:
Sure! Here's a Go blog post for the `go fmt` tool: https://go.dev/blog/gofmt

The tool is very simple to use manually. For example you can run on your terminal `go fmt path/to/your/file.go`.

If you are using an editor such as VSCode, you may use the official Go plugin to get automatic formatting built-in: https://code.visualstudio.com/docs/languages/go#_formatting